pkg/policy/l4.go

```diff
@@ -39,13 +40,25 @@ func (l7 L7DataMap) MarshalJSON() ([]byte, error) {
 		return []byte("[]"), nil
 	}
 
+	/* First, create a sorted slice of the selectors so we can get
+	 * consistent JSON output */
+	selectors := make(api.EndpointSelectorSlice, len(l7))
+	for es := range l7 {
+		selectors = append(selectors, es)
+	}
+	sort.Sort(selectors)
+
+	/* Now we can iterate the slice and generate JSON entries. */
 	var err error
 	buffer := bytes.NewBufferString("[")
-	for k, v := range l7 {
+	for _, es := range selectors {
+		if _, ok := l7[es]; !ok {
+			continue
+		}
 		buffer.WriteString("\n  {    \n    \"")
-		buffer.WriteString(k.LabelSelectorString())
+		buffer.WriteString(es.LabelSelectorString())
 		buffer.WriteString("\": ")
-		b, err := json.MarshalIndent(v, "    ", "  ")
```

Comment:
This line is unnecessary, as well all the `\n` and spaces since this function is `MarshalJSON`, not `MarshalIndent`

https://golang.org/pkg/encoding/json/#MarshalIndent

Reply:
Fixed.